subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/StrictDependenciesIntegrationTest.groovy

```diff
@@ -15,30 +15,26 @@
  */
 package org.gradle.integtests.resolve
 
-import org.gradle.integtests.fixtures.AbstractHttpDependencyResolutionTest
-import org.gradle.integtests.fixtures.executer.GradleContextualExecuter
-import org.gradle.integtests.fixtures.resolve.ResolveTestFixture
 import spock.lang.Unroll
 
-class StrictDependenciesIntegrationTest extends AbstractHttpDependencyResolutionTest {
-    def resolve = new ResolveTestFixture(buildFile, "conf")
-
-
-    def setup() {
-        settingsFile << "rootProject.name = 'test'"
-        resolve.prepare()
-        server.start()
-    }
-
+class StrictDependenciesIntegrationTest extends AbstractStrictDependenciesIntegrationTest {
 
     void "can declare a strict dependency onto an external component"() {
         given:
-        def m = mavenHttpRepo.module("org", "foo", '1.0').publish()
+        repository {
+            group("org") {
+                module("foo") {
+                    version("1.0") {
+                        expectGetMetadata()
+                        expectGetArtifact()
+                    }
+                }
+            }
+        }
```

Comment:
I'm not 100% convinced that this is better, due to the verbosity. But I'm willing to see how it evolves.
I guess there's no reason we can't have a fluent version for the simple use cases (like this one).

Reply:
It is strictly equivalent to using this syntax (which already works):

```
        repository {
            'org:foo:1.0' {
                expectGetMetadata()
                expectGetArtifact()
            }
        }

```

The difference is that the 1st one (long version) has IDE completion. The 2d doesn't.